io.openems.edge.evcs.ocpp.server/src/io/openems/edge/evcs/ocpp/server/EvcsOcppServer.java

```diff
@@ -110,21 +110,33 @@ protected void removeEvcs(Evcs evcs) {
 			return;
 		}
 		var ocppEvcs = (AbstractManagedOcppEvcsComponent) evcs;
-		var evcss = this.activeEvcsSessions.get(ocppEvcs.getSessionId());
-		if (evcss != null) {
-			if (evcss.size() < 2) {
-				this.activeEvcsSessions.remove(ocppEvcs.getSessionId());
-			} else {
-				this.activeEvcsSessions.get(ocppEvcs.getSessionId()).remove(ocppEvcs);
+		var sessionId = ocppEvcs.getSessionId();
+		if (sessionId != null) {
+			var evcss = this.activeEvcsSessions.get(sessionId);
+			if (evcss != null) {
+				evcss.remove(ocppEvcs);
+				if (evcss.isEmpty()) {
+					this.activeEvcsSessions.remove(sessionId);
+					// Also remove from ocppSessions if needed
```

Comment:
In actual production environments, we rarely remove components, so we went ahead with weekend testing without including this commit. However, I don’t see any issues with the code itself.